### ISSUE TITLE
Pasting multi line text

### ISSUE THREAD

**Issue by PEsteves8** [mui/mui-x#11754]
### Search keywords

paste pasting

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: [(required)](https://mui.com/x/react-data-grid/clipboard/) (just paste a multiline text in a cell

You have an example of turning a text cell into a multiline text box. However when copying and pasting from this cell, the contents get spread to multiple cells vertically

Steps:
1. Copy a multiline text
2. Paste it into a text cell
3. The text content will be spread vertically in multiple cells

### Current behavior

Multiline text gets spread vertically in multiple text cells.

Also it includes quote marks.

### Expected behavior

Multiline text gets pasted into one cell.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID or Support key 💳 (optional)

82566

**Comment by michelengelen** [comment]
I am not sure I follow the problem you are experiencing. I can copy and paste multi-line text just fine in [this example](https://mui.com/x/react-data-grid/recipes-editing/#multiline-editing).

https://github.com/mui/mui-x/assets/32863416/7cacf048-0f2a-4108-9507-789a03168790

Could you maybe record a short video of the problem you are experiencing? Thanks! 🙇🏼

**Comment by PEsteves8** [comment]
Ah I can now see that it is specific to DataGridPremium and to version 7-alpha.9 (don't know it it applies to previous alphas).

If you use latest alpha with DataGridPremium, you'll see that pasting doesn't work at all.

It's different than what I was experiencing but perhaps related (In my case, each line of the multiline text would be pasted to each subsequent cell below the cell on which the text was being pasted)

**Comment by michelengelen** [comment]
Well... IMHO it is exactly working like expected.
I did run some experiments and this is what I found:
if you have clipboardPaste enabled (this is the default and only available on the premium) you can copy/paste full rows or cells and paste them back, regardless of the selection mode. Now for the multi-line text: this one will paste into multiple cells when you have either `disableRowSelectionOnClick` set to true, since this will disable the selection of a full row and therefore allow for pasting in multi-line text that get spread onto the cells.
If you set `disableClipboardPaste` to true it will paste the multi-line text into a single cell (the one selected with the focus border)

TLDR;
The behavior is as expected, its just unintuitive.

I will keep this open and put it onto our board to improve the docs in this regard. Thanks 🙇🏼

**Comment by PEsteves8** [comment]
Aaaaah I see. Ok that makes sense. Just wish the ctrl + z (https://github.com/mui/mui-x/issues/9339) feature was already there, as I can imagine users pasting multiline text and overwriting cells non intentionally, and then wanting to undo and actual open the edit mode of the multiline cell to paste. :)

Thanks!

**Comment by PEsteves8** [comment]
I can see that Airtable handles this a bit better.

So here's an example:

<img width="95" alt="image" src="https://github.com/mui/mui-x/assets/16803226/dd06dd51-2657-4ea2-8dbb-e225d11b45d6">


the cell with the numbers is actually multiline, with a number on each line. If I copy these two cells, the clipboard will have this content:

`"1\n2\n3\n4\n5"\nA`

The double quotes are part of the content. 

If I paste this in a multiline cell of the mui-x grid, the double quotes will be included, although they didn't exit in the original cell.

Airtable seems to look at the double quotes and assume that the content inside of them belongs in one cell, so it does paste the numbers in one cell, and the letter in the cell below it.
